torchrl/collectors/collectors.py

```diff
@@ -124,6 +128,28 @@ def _get_policy_and_device(
                     "env must be provided to _get_policy_and_device if policy is None"
                 )
             policy = RandomPolicy(self.env.action_spec)
+        elif (
+            observation_spec is not None
+            and isinstance(policy, nn.Module)
+            and not isinstance(policy, TensorDictModule)
+        ):
+            # auto-wrap policy with TensorDictModule
+            sig = inspect.signature(policy.forward)
+            next_observation = {
+                key[5:]: value
+                for key, value in observation_spec.rand().items()
```

Comment:
no need for rand, we should be good with `observation_spec.items()`

Reply:
The reason for putting `rand` in there was because a couple of lines down I want to do `output = policy(**next_observation)` to count output arguments and if I attempt to do this just on the spec items I get errors like

```
TypeError: linear: argument 'input' (position 1) must be Tensor, not NdBoundedTensorSpec
```